toucan_connectors/toucan_connector.py

```diff
@@ -61,36 +60,40 @@ class DataSlice(NamedTuple):
 
     df: pd.DataFrame  # the dataframe of the slice
     # TODO total_count field should be removed
-    total_count: Optional[int] = None  # the length of the raw dataframe (without slicing)
-    input_parameters: Optional[dict] = None
-    stats: Optional[DataStats] = None
+    total_count: int | None = None  # the length of the raw dataframe (without slicing)
+    input_parameters: dict | None = None
+    stats: DataStats | None = None
     # TODO: name is kinda misleading. what others information than `columns` will it contain ?
-    query_metadata: Optional[QueryMetadata] = None
+    query_metadata: QueryMetadata | None = None
 
 
 class StrEnum(str, Enum):
     """Class to easily make schemas with enum values and type string"""
 
 
-def strlist_to_enum(field: str, strlist: List[str], default_value=...) -> tuple:
+def strlist_to_enum(field: str, strlist: list[str], default_value=...) -> tuple[type[StrEnum], Any]:
     """
     Convert a list of strings to a pydantic schema enum
     the value is either <default value> or a tuple ( <type>, <default value> )
     If the field is required, the <default value> has to be '...' (cf pydantic doc)
     By default, the field is considered required.
     """
-    return StrEnum(field, {v: v for v in strlist}), default_value
+    return StrEnum(field, {v: v for v in strlist}), default_value  # type: ignore[call-overload]
 
 
-class ToucanDataSource(BaseModel):
+# Binding C to ToucanConnector causes mypy to crash (last checked with mypy 0.991)
```

Comment:
Is there a referal url for this issue yet ? if yes, providing the link in this comment could be great IMO !

Reply:
Nope, guess mutually-referencing GATs is an edge case for mypy :sweat_smile: